placement/selector/mirrored.go

```diff
@@ -358,10 +358,26 @@ func assignShardsToGroupedInstances(
 		instances   = make([]placement.Instance, 0, p.ReplicaFactor()*len(groups))
 		shardSetIDs = nextNShardSetIDs(p, len(groups))
 	)
-	for i, group := range groups {
-		ssID := shardSetIDs[i]
+	for _, group := range groups {
+		var (
+			useNewSSID bool
+			ssID       uint32
+		)
+		for _, instance := range group {
+			_, exist := p.Instance(instance.ID())
+			if !exist {
+				useNewSSID = true
```

Comment:
Do we want to validate that all of them (instead of only a subset of them) are in the group (or not in the group)?

Reply:
I think we should, actually I think the check should be more strict so that all the instances in the same group needs to have the same shard set id to avoid be assigned a new one.